ntp-udp/src/raw_socket.rs

```diff
@@ -27,62 +27,101 @@ pub(crate) fn cerr(t: libc::c_int) -> std::io::Result<libc::c_int> {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+#[repr(i32)]
+#[allow(clippy::enum_variant_names)]
+pub(crate) enum TimestampMethod {
+    #[allow(dead_code)]
+    SoTimestamp = libc::SO_TIMESTAMP,
+    #[cfg(target_os = "linux")]
+    #[allow(dead_code)]
+    SoTimestampns = libc::SO_TIMESTAMPNS,
```

Comment:
Should we even keep this around? There seems to be no case that uses this which risks this becoming unusable fairly quickly.

Reply:
ntpsec and chrony still have this so :man_shrugging: We do run a test with it, and it doesn't really cost anything to support it. My guess is that there are linux versions where `SO_TIMESTAMPNS` is available but `SO_TIMESTAMPING` is not. Or maybe `SO_TIMESTAMPNS` is better when you don't use send timestamps?

Reply:
Ok, let's keep it in